download_and_create_reference_datasets/v02/mito/utils.py

```diff
@@ -0,0 +1,102 @@
+        raise ValueError("Invalid url: {}".format(url))
+
+    local_file_path = os.path.join(to_dir, os.path.basename(url.rstrip('/')))
+
+    if verify != False:
+        response = requests.head(url)
+        size = int(response.headers.get('Content-Length', '0'))
+        if os.path.isfile(local_file_path) and os.path.getsize(local_file_path) == size:
+            logger.info("Re-using {} previously downloaded from {}".format(local_file_path, url))
+            return local_file_path
+
+    is_gz = url.endswith(".gz") or url.endswith(".zip")
+    response = requests.get(url, stream=is_gz, verify=True if verify is None else verify)
+    input_iter = response if is_gz else response.iter_content()
+
+    logger.info("Downloading {} to {}".format(url, local_file_path))
+    input_iter = tqdm.tqdm(input_iter, unit=" data" if is_gz else " lines")
+
+    with open(local_file_path, 'wb') as f:
+        f.writelines(input_iter)
+
+    input_iter.close()
+
+    return local_file_path
+
+
+def convert_json2tsv(json_path):
+    with open(json_path, 'r') as f:
+        data = json.load(f)
+    tsv_path = f'{json_path[:-5]}.tsv' if json_path.endswith('.json') else f'{json_path}.tsv'
+    with open(tsv_path, 'w') as f:
+        header = '\t'.join(data[0].keys())
+        f.write(header + '\n')
+        for row in data:
+            f.write('\t'.join([str(v) for v in row.values()]) + '\n')
+    return tsv_path
+
+
+def load_ht(config, force_write=True):
+    logger.info(f'Downloading dataset from {config["input_path"]}.')
+    dn_path = download_file(config['input_path'], verify=config.get('verify_ssl'))
+
+    if dn_path.endswith('.zip'):
+        with zipfile.ZipFile(dn_path, 'r') as zip:
+            zip.extractall(path=os.path.dirname(dn_path))
+        dn_path = dn_path[:-4]
+
+    logger.info(f'Loading hail table from {dn_path}.')
+    types = config['field_types'] if config.get('field_types') else {}
+    if config['input_type'] == 'vcf':
+        ht = hl.import_vcf(dn_path, force_bgz=True, contig_recoding=contig_recoding).rows()
+    elif config['input_type'] == 'json':
+        tsv_path = convert_json2tsv(dn_path)
+        ht = hl.import_table(tsv_path, types=types)
+    else:
+        ht = hl.import_table(dn_path, types=types)
+
+    if config.get('annotate'):
+        ht = ht.annotate(**{field: func(ht) for field, func in config['annotate'].items()})
+
+    ht = ht.filter(ht.locus.contig == 'chrM')
+
+    ht = ht.key_by('locus', 'alleles')
+
+    logger.info(f'Writing hail table to {config["output_path"]}.')
+    ht.write(config['output_path'], overwrite=force_write)
+    logger.info('Done')
+
+
+def load(config):
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-f', '--force-write', help='Force write to an existing output file', action='store_true')
+    args = parser.parse_args()
+
+    # If there are out-of-memory error, set the environment variable with the following command
```

Comment:
this seems like its copy-pasted from somewhere - if so, can you add a link to the source in the comment? It means in the future its easier to check and see if a workaround is still necessary/ recommended

Reply:
It is about the memory size setting for Hail. I occasionally found the environment variable PYSPARK_SUBMIT_ARGS setting in a [forum](https://discuss.hail.is/t/java-heap-space-out-of-memory/1575/6). I can't find it in Hail docs. So, I doc it here to remind whoever may need it later.

Reply:
cool, so just add the link to that forum in the comment here

Reply:
I found that it doesn't require any special settings to import each specific mito reference dataset. So, I removed the comments and the corresponding parameters for `hl.init()`. But the settings are still required for combining all reference datasets.